### PR TITLE
remove primary_for_device in data migration

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #4104 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/4104.removed [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove leftover `primary_for_device`  from interface data migration.
```
Comment (HanlinMiao) [inline changes/4104.removed:1]:
```suggestion
Remove leftover `primary_for_device` from Interface data migration.
```

**Review by HanlinMiao — APPROVED** [review]
Flake8 is failing but otherwise looks good

**Review by jathanism — APPROVED** [review]

File: nautobot/ipam/migrations/0025_interface_ipaddress_m2m_data_migration.py [hunk 2]
```diff
@@ -14,31 +13,21 @@ def migrate_ipaddress_to_m2m(apps, schema_editor):
         related_ct = ip_address.assigned_object_type
         if related_ct.app_label == "dcim" and related_ct.model == "interface":
             related_obj = Interface.objects.get(id=ip_address.assigned_object_id)
-            parent = Device.objects.get(id=related_obj.device_id)
-            m2m = IPAddressToInterface.objects.create(
+            Device.objects.get(id=related_obj.device_id)
```
Comment (glennmatthews) [inline nautobot/ipam/migrations/0025_interface_ipaddress_m2m_data_migration.py:16]: Probably just delete this line?

```suggestion
```

File: nautobot/ipam/migrations/0025_interface_ipaddress_m2m_data_migration.py [hunk 3]
```diff
@@ -14,31 +13,21 @@ def migrate_ipaddress_to_m2m(apps, schema_editor):
...
+            IPAddressToInterface.objects.create(
                 ip_address=ip_address,
                 interface=related_obj,
             )
         elif related_ct.app_label == "virtualization" and related_ct.model == "vminterface":
             related_obj = VMInterface.objects.get(id=ip_address.assigned_object_id)
-            parent = VirtualMachine.objects.get(id=related_obj.virtual_machine_id)
-            m2m = IPAddressToInterface.objects.create(
+            VirtualMachine.objects.get(id=related_obj.virtual_machine_id)
```
Comment (glennmatthews) [inline nautobot/ipam/migrations/0025_interface_ipaddress_m2m_data_migration.py:23]:
```suggestion
```

File: changes/4104.removed [hunk 4]
```diff
@@ -0,0 +1 @@
+Remove leftover `primary_for_device` from Interface data migration.
```
Comment (gsnider2195) [inline changes/4104.removed:1]:
```suggestion
Removed unused `primary_for_device` from Interface data migration.
```

**Comment by gsnider2195** [comment]
@timizuoebideri1 I fixed the flake8 failures for you since I had your branch checked out to review the changes

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**